test/integration/targets/git/tasks/submodules.yml

```diff
@@ -14,6 +14,90 @@
 # Repository III for a second submodule (repo_submodule2)
 #   Has 1 file checked in
 
+- name: SUBMODULES | setup
+  block:
+    - name: SUBMODULES | setup | start git daemon and create repositories
+      include_role: name=test_utils/roles/git-repos
+      vars:
+        repositories:
+          - test_submodules_newer.git
+          - test_submodules_subm1.git
+          - test_submodules_subm2.git
+        exec:
+          - 'echo initial > README.md'
+          - 'git add README.md && git commit -m "Initial commit"'
+          - 'git tag v0.1'
+          - 'git push origin {{ git_default_branch }} --tags'
+
+    - name: SUBMODULES | setup | add submodule1
+      include_role: name=test_utils/roles/git-repos
+      vars:
+        repositories:
+          - test_submodules_newer.git
+        exec:
+          - 'git submodule add --name submodule1 git://localhost/test_submodules_subm1.git submodule1'
+          # workaround for changing default submodule branch name
```

Comment:
Can this be done through a `.gitmodules` file instead?

Reply:
This does create .gitmodules.